test/suite/test_sweep06.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_sweep06.py
+# This test confirms that table dhandles are correctly retained and not
+# marked for sweep server expiry when file data dhandles are present.
+
+from suite_subprocess import suite_subprocess
+import wttest, threading
+
+class test_sweep06(wttest.WiredTigerTestCase, suite_subprocess):
+    dhandles = 1000
+    format='key_format=i,value_format=S'
+    tablebase = 'test_sweep06'
+    uri = 'table:' + tablebase
+    conn_config = 'file_manager=(close_handle_minimum=0,' + \
+                  'close_idle_time=60,close_scan_interval=30),'
+
+    def insert(self, i, start, rows):
+        session = self.conn.open_session()
+        uri = self.uri + str(i)
+        cursor = session.open_cursor(uri)
+        session.begin_transaction()
+        for i in range(start, rows):
+            cursor.set_key(i)
+            cursor.set_value(str(i))
+            cursor.insert()
+        session.commit_transaction()
+        cursor.close()
+        session.close()
+
+    def test_dhandles(self):
+        for i in range(1,self.dhandles):
+            uri = self.uri + str(i)
+            self.session.create(uri, self.format)
+
+        for i in range(1,100):
+            threads = []
+            for i in range(1,self.dhandles):
+                thread = threading.Thread(target=self.insert, args=(i, 0, 100))
+                thread.start()
+                threads.append(thread)
+
+            for thread in threads:
+                thread.join()
```

Comment:
It'll be useful if we add some comments and stats to check the expected behavior?